script/get-visual-studio-verion.ps1

```diff
@@ -0,0 +1 @@
+(vswhere -products * -find **/Hostx86/x86/* | select-string '(\\MSVC\\)(.*)(\\bin)').matches.groups[2].value
```

Comment:
This requires vswhere which may not be installed. I was recently involved with a project that uses some manual code to find Visual Studio in case `vswhere` is not installed.

https://github.com/ilammy/msvc-dev-cmd/blob/1cccac6a8dab508b878aeb0f56ff9b2ca774d201/index.js#L52-L64.

Reply:
Another possibility is to bundle `vswhere` with version tracker. It is a tiny exe file:
https://github.com/microsoft/vswhere/releases/download/2.8.4/vswhere.exe

Reply:
dang, all that is needed just to check the version? 😬  I guess storing the exe in the atom repo is the best option

Reply:
I'd suggest that you add this to the version tracker itself. It makes sense that you right a little script that downloads it and runs it. That would be useful for anyone who uses the version tracker (not just Atom).

Here is a sketch. Should I make a PR?
```js

const fs = require('fs');
const path = require('path');
const download = require('download');
const childProcess = require('child_process');

const vswhereVersion = '2.8.4';

const PROGRAM_FILES_X86 = process.env['ProgramFiles(x86)'] || 'ProgramFiles(x86)'
const vswhereDir = `${PROGRAM_FILES_X86}\\Microsoft Visual Studio\\Installer`
const vswherePath = `${vswhereDir}\\vswhere.exe`

async function downloadVsWhere() {
  if (!fs.existsSync(vswherePath)) {
    await download(
      `https://github.com/microsoft/vswhere/releases/download/${vswhereVersion}/vswhere.exe`,
      vsWherePath,
    );
  }
}

function findWithVswhere(pattern) {

	// if already installed and on PATH
    try {
        let installationPath = child_process.execSync(`vswhere -products * -latest -prerelease -property installationPath`).toString().trim()
        return installationPath + '\\' + pattern
    } catch (e) {
        console.warn(`vswhere failed: ${e}`)
    }

	// download
	await downloadVsWhere() 

    try {
        let installationPath = child_process.execSync(`${VSWHERE} -products * -latest -prerelease -property installationPath`).toString().trim()
        return installationPath + '\\' + pattern
    } catch (e) {
        console.warn(`vswhere failed: ${e}`)
    }

    return null
}
```

Reply:
I can make another repo that wraps that up, but I don't want to hardcode anything for version-tracker. I don't want any maintenance, just one-and-done generic tool that reusable and has 0-dependencies. Which is also why I didn't use `shelljs` 

For the wrapper repo, since its going to need maintenance can we add it under atom-ide-community? (so all the contributors can merge PR's)

Reply:
What I'm confused about is: is there no global CLI for visual studio? It looks like you can have multiple versions of visual studio installed, but in that case which one is atom going to use? Latest makes the most sense, but it seems like you can also have Enterprise and other variants installed

Reply:
I also kinda think the version tracking would be useful as-is, its just a debugging tool for people who build atom. We can mention the vswhere check+download in the docs, and create an issue to keep track of it, but the optional dependency of Visual Studio is kinda blocking/slowing an otherwise straightforward improvement.

If vswhere is included in all Visual Studio versions after 2017, I say we just require Visual Studio ≥ 2017